### PR TITLE
New widgets architecture + fixes

### PR DESCRIPTION
**Pull request by Bars92**
support new widgets architecture + bar chart legend fix + increased pie labels offset

example:
image

### CODE REVIEW

File: src/components/Sections/SectionChart/ChartLegend.js [hunk 1]
```diff
@@ -30,24 +30,20 @@ const ChartLegend = ({ data, icon = 'circle', layout = CHART_LAYOUT_TYPE.vertica
...
     const legendIconClass = `${icon} icon chart-legend-icon`;
     let width = 'auto';
-    let maxWidth;
     const value = group.value ? numberToShortString(group.value) : null;
     // decrease width of name (if value exists) to allow for ellipsis.
     if (value && showValue) {
       let valueInPixels = (value + '').replace('.', '').length * DIGIT_PIXEL_SIZE + ICON_CONTAINER_PIXEL_SIZE;
       if (valueDisplay === VALUE_FORMAT_TYPES.stretch) {
         width = `calc(100% - ${valueInPixels}px)`;
-      } else if (valueDisplay === VALUE_FORMAT_TYPES.minimal) {
```
Comment (orius123) [inline src/components/Sections/SectionChart/ChartLegend.js]: we don't support it anymore?

File: src/components/Sections/SectionChart/SectionLineChart.js [hunk 2]
```diff
@@ -38,11 +38,23 @@ const SectionLineChart = ({ data, style, dimensions, legend, chartProperties = {
...
-        Object.keys(mainGroup).filter(key => key !== 'name' && key !== 'relatedTo').forEach(groupKey => {
-          lineTypes[groupKey] = { name: groupKey, stroke: getGraphColorByName(groupKey) };
-        });
+        const mainObject = { name };
+        if (mainGroup.groups && mainGroup.data) {
+          // add all sub groups to main object.
+          mainGroup.groups.forEach((group) => {
+            const groupName = group.name;
+            const id = group.name;
+            mainObject[groupName] = group.data[0];
```
Comment (orius123) [inline src/components/Sections/SectionChart/SectionLineChart.js]: `group.data` will always have a value?

Comment (Bars92) [reply src/components/Sections/SectionChart/SectionLineChart.js]: yeah always. can be `0`

File: src/components/Sections/SectionChart/ChartLegend.js (hunk 1 above)
Comment (Bars92) [reply src/components/Sections/SectionChart/ChartLegend.js]: not needed for reports. just causing issues, ellipsis not needed

**Pull request merged**